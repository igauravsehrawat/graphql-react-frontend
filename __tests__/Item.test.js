import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'abc',
  image: 'dog.jpg',
  largeImage: 'largeDog.jpg',
  price: 100,
  description: 'This is dummy description',
  title: 'This is title here',
};

describe('Item', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders img src and alt', () => {
    const rendered = shallow(<ItemComponent item={fakeItem} />);
    const img = rendered.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });
  it('renders title and price tag', () => {
    const rendered = shallow(<ItemComponent item={fakeItem} />);
    const priceTag = rendered.find('PriceTag');
    expect(priceTag.children().text()).toBe('â‚¬1');
    expect(rendered.find('Title Link a').text()).toBe(fakeItem.title);
  });

  it('renders out the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link')).toHaveLength(1);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });
});
