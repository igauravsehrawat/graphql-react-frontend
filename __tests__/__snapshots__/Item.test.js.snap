// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders and matches the snapshot 1`] = `
<div>
  <ItemStyles__Item>
    <img
      alt="This is title here"
      src="dog.jpg"
    />
    <Title>
      <Link
        href={
          Object {
            "pathname": "/item",
            "query": Object {
              "id": "abc",
            },
          }
        }
      >
        <a>
          This is title here
        </a>
      </Link>
    </Title>
    <PriceTag>
      €1
    </PriceTag>
    <p>
      This is dummy description
    </p>
    <div
      className="buttonList"
    >
      <Link
        href={
          Object {
            "pathname": "/update",
            "query": Object {
              "id": "abc",
            },
          }
        }
      >
        <button>
          Edit ✏️
        </button>
      </Link>
      <AddToCart
        id="abc"
      />
      <DeleteItem
        id="abc"
      >
        Delete this item
      </DeleteItem>
    </div>
  </ItemStyles__Item>
</div>
`;
